### PR TITLE
Km451 refactor helm client

### PR DESCRIPTION
**Pull request by ivaruf**
Remove direct helm client. Init state directly with helm service.

### CODE REVIEW

File: pkg/client/core/state/direct/blockstorage_state.go [hunk 1]
```diff
@@ -44,9 +45,9 @@ func (b blockstorageState) HasBlockstorage() (bool, error) {
 }
 
 // NewBlockstorageState returns an initialized state client
-func NewBlockstorageState(clusterMeta v1alpha1.ClusterMeta, helmClient client.HelmAPI) client.BlockstorageState {
+func NewBlockstorageState(clusterMeta v1alpha1.ClusterMeta, helhelmService api.HelmService) client.BlockstorageState {
```
Comment (ivaruf) [inline pkg/client/core/state/direct/blockstorage_state.go:48]: Typo

**Review by yngvark — APPROVED** [review]
👍

**Review by deifyed — APPROVED** [review]
I've written a few comments about the `context.Background()` usage. Since we arent actually using this feature, I won't require any changes. However, I'm afraid we'll kick ourself in the foot later on when these chains are broken. It's just so easy to do it right the first time IMO

File: pkg/client/core/state/direct/autoscaler_state.go [hunk 2]
```diff
@@ -22,7 +23,7 @@ func (a autoscalerState) HasAutoscaler() (bool, error) {
 		0*time.Minute,
 	)
 
-	_, err := a.helm.GetHelmRelease(api.GetHelmReleaseOpts{
+	_, err := a.helm.GetHelmRelease(context.Background(), api.GetHelmReleaseOpts{
```
[the remark this answers is not in the document]
Comment (ivaruf) [reply pkg/client/core/state/direct/autoscaler_state.go:26]: Trying to not change too many signatures this time around. But, maybe I should make a separate walk-through the code when I am done with direct clients - and make sure we send context around (basically freetext search for context.Background and take it from there.)

[the remark this answers is not in the document]
Comment (deifyed) [reply pkg/client/core/state/direct/autoscaler_state.go:26]: yeah! sounds like a plan. If it doesn't take too long. Like I said, its not currently in use. I just envision the frustration when all the chains are broken

**Pull request merged**